File: Src/java/engine-jackson/src/test/java/org/opencds/cqf/cql/engine/execution/Issue39.java
package org.opencds.cqf.cql.engine.execution;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;

import org.testng.annotations.Test;

public class Issue39 extends CqlExecutionTestBase {
    @Test
    public void testInterval() {
        Context context = new Context(library);
        Object result = context.resolveExpressionRef("EquivalentIntervals").getExpression().evaluate(context);
        assertThat(result, is(true));
    }
}

File: Src/java/engine/src/jvmTest/kotlin/org/opencds/cqf/cql/engine/execution/Issue39.kt
package org.opencds.cqf.cql.engine.execution

import kotlin.test.Test
import kotlin.test.assertEquals
import org.opencds.cqf.cql.engine.runtime.Boolean

internal class Issue39 : CqlTestBase() {
    @Test
    fun interval() {
        val results = engine.evaluate { library("Issue39") }.onlyResultOrThrow
        val value = results["EquivalentIntervals"]!!.value
        assertEquals(Boolean.TRUE, value)
    }
}
